core/src/helpers/parseMetamaskError.ts

```diff
@@ -1,15 +1,8 @@
 import truncateText from './truncateText';
 
-const parseMetamaskError = function (errorMessage = ''): string {
-    const jsonFirstIndex = errorMessage.indexOf('{');
-    const jsonLastIndex = errorMessage.lastIndexOf('}');
-    try {
-        const jsonString = errorMessage.substring(jsonFirstIndex, jsonLastIndex + 1);
-        const metamaskError = JSON.parse(jsonString);
-        return truncateText(metamaskError?.value?.data?.message || 'unknown');
-    } catch {
-        return truncateText(errorMessage || 'unknown');
-    }
+const parseMetamaskError = function (errorMessage = ''): unknown {
+    const splitError = errorMessage.split('(')[0];
+    return truncateText(splitError);
```

Comment:
This will display only the first part of the error, while the most important one, as already said in the issue and in the previous PR would be the part which is inside `value.data.message`. As suggested before the message displayed to the user should be `${metamaskError.value.data.message} (${metamaskErrorMessage})`. But you currently only display `metamaskErrorMessage`